src/shop-app.html

```diff
@@ -629,16 +631,25 @@
         window.setTimeout(function() {
           instrumentResponse.complete('success')
             .then(function() {
-              if (!that._orderSuccessModal) {
-                that._orderSuccessModal = document.createElement('shop-order-success-modal');
-                Polymer.dom(that.root).appendChild(that._orderSuccessModal);
-              }
-              that._orderSuccessModal.open();
-              that._announce('Order processed successfully!');
-
               if (cartBuy) {
                 that.fire('clear-cart');
               }
+
+              that.accountData = that.$.meta.byKey('account');
+
+              if (that.accountData.isSignedIn()) {
+                if (!that._orderSuccessModal) {
+                  that._orderSuccessModal = document.createElement('shop-order-success-modal');
+                  Polymer.dom(that.root).appendChild(that._orderSuccessModal);
+                }
+                that._orderSuccessModal.open();
+              } else {
+                that.fire('route-change', {
+                  path: '/account'
```

Comment:
Looks like L652 addresses the same issue? In that case, I will need to remove the similar bit on `shop-account.html`.